FWCore/ParameterSet/python/Config.py

```diff
@@ -1278,25 +1278,58 @@ def __call__(self,obj):
   def _raiseUnknownKey(key):
     raise KeyError("Unknown parameter name "+key+" specified while calling Modifier")
 
-class _AndModifier(object):
-  """A modifier which only applies if multiple Modifiers are chosen"""
-  def __init__(self, lhs, rhs):
-    self.__lhs = lhs
-    self.__rhs = rhs
-  def isChosen(self):
-    return self.__lhs.isChosen() and self.__rhs.isChosen()
+class _BoolModifierBase(object):
+  """A helper base class for _AndModifier, _InvertModifier, and _OrModifier to contain the common code"""
+  def __init__(self, lhs, rhs=None):
+    self._lhs = lhs
+    if rhs is not None:
+      self._rhs = rhs
   def toModify(self,obj, func=None,**kw):
+    Modifier._toModifyCheck(obj,func,**kw)
+    if not self.isChosen():
+      return
+    self._toModify(obj,func,**kw)
+  def _toModify(self,obj,func,**kw):
+    self._lhs._toModify(obj,func, **kw)
```

Comment:
I assume you never call `self._rhs._*` because you assume the `_*` does the exact same work for `_lhs` and '_rhs`? That is because you ultimately expect that to call `Modifier._*`?

Would it be better to make `Modifier._*` a `@staticmethod` and all it directly?

Reply:
> I assume you never call self._rhs._* because you assume the _* does the exact same work for _lhs and 'rhs? That is because you ultimately expect that to callModifier.*`?

Yes, your assumption is correct.

> Would it be better to make Modifier._* a @staticmethod and all it directly?

I agree it would make the chain simpler to understand (and probably a bit faster as well since one doesn't have to trace the contained Modifier tree back to some leaf).